packages/quantic/force-app/main/default/classes/InsightController.cls

```diff
@@ -0,0 +1,20 @@
+/**
+ * Simple Apex Controller to fetch data for Headless initialization specific to the Insight use case.
+ */
+global with sharing class InsightController {
+    /**
+     * Get the configuration object (serialized) for the initialization of Coveo Headless through a class implementing `ITokenProvider`.
+     * @return {String} A serialized object containing the following keys:
+     *                      - `platformUrl`: the endpoint to contact Coveo.
+     *                      - `accessToken`: an access token (see https://docs.coveo.com/en/123/apis/get-your-coveo-access-token).
+     *                      - `organizationId`: the ID of the organization to query (see https://docs.coveo.com/en/148/manage-an-organization/retrieve-the-organization-id).
+     */
+    @AuraEnabled(cacheable=true)
+    public static String getHeadlessConfiguration() {
+        /**
+         * e.g. To switch to the Insight Token Provider, you simply need to replace the following line by:
+         * `return InsightTokenProvider.getHeadlessConfiguration();`
```

Comment:
@mmitiche `InsightTokenProvider.getHeadlessConfiguration()` is the one that links Quantic to the Coveo for Salesforce package. It is Coveo for Salesforce that will expose a method providing the scoped platform token.

We'll then be able to add `InsightTokenProvider` to Quantic.

Reply:
Understood 👌 Merging this [Jira](https://coveord.atlassian.net/browse/SFINT-4491) will allow us to expose the method providing the platform token in C4SF. Will try to merge it soon!

Reply:
The method providing the scoped platform token is now exposed in the C4SF package, we can now proceed and create the InsightTokenProvider.

Reply:
The `InsightTokenProvider` class has been created.